src/services/ManualMapLoading.ts

```diff
@@ -0,0 +1,207 @@
+    }
+    const envir = file.match(/desc envir=".*?"/gm)?.[0].slice(12, -1)
+    if (this.stadium && envir !== "Stadium") {
+      return new Error("MISMATCH: Map " + uid + " has environment " + envir + ", not Stadium")
+    }
+    const author = file.match(/" author=".*?"/gm)?.[0].slice(10, -1).slice(0, 40)
+    // Yes, author logins can sometimes be longer than 40 characters and map names can be longer than 60
+    // ...for some reason (thanks Nadeo). Cut them here to prevent errors later.
+    const name = file.match(/" name=".*?"/gm)?.[0].slice(8, -1).slice(0, 60)
+    const price = file.match(/price=".*?"/gm)?.[0].slice(7, -1)
+    const goldTime = file.match(/gold=".*?"/gm)?.[0].slice(6, -1)
+    const mood = file.match(/["\x00](Day|Night|Sunrise|Sunset)/gm)?.[0].slice(1)
+    const bronzeTime = file.match(/bronze=".*?"/gm)?.[0].slice(8, -1)
+    const silverTime = file.match(/silver=".*?"/gm)?.[0].slice(8, -1)
+    const authorTime = file.match(/authortime=".*?"/gm)?.[0].slice(12, -1)
+    const nbLaps = file.match(/nblaps=".*?"/gm)?.[0].slice(8, -1)
+    if (uid == undefined || author == undefined || name == undefined || envir == undefined) {
+      return new Error("PARSEERROR: Could not get map info of file " + filename)
+    }
+    return {
+      Name: name,
+      UId: uid,
+      FileName: filename,
+      Environnement: envir,
+      Author: author,
+      GoldTime: goldTime == undefined? 0 : parseInt(goldTime),
+      CopperPrice: price == undefined? 0 : parseInt(price),
+      Mood: mood == undefined? "Day" : mood,
+      BronzeTime: bronzeTime == undefined? 0 : parseInt(bronzeTime),
+      SilverTime: silverTime == undefined? 0 : parseInt(silverTime),
+      AuthorTime: authorTime == undefined? 0 : parseInt(authorTime),
+      NbLaps: nbLaps == undefined? 0 : parseInt(nbLaps)
+    }
+  }
+
+  /**
+   * Create a MatchSettings file from the queue and load it into the server.
+   * Does not write the file if it would have not changed.
+   * WARNING: If the queue contains invalid maps (e.g. puzzles) the server keeps the old MatchSettings.
+   * @param curr the current map
+   * @param queue the queue
+   * @param startAt
+   */
+  static async writeMS(curr: tm.CurrentMap, queue: tm.Map[], startAt: number = 0) {
+    const newQueue = (queue.slice(0, config.preloadMaps))
+    if (this.oldQueue !== undefined && this.oldCurr !== undefined
+      && curr.id === this.oldCurr.id && curr.fileName === this.oldCurr.fileName &&
+      this.oldQueue.every(((a, i) => a.id === newQueue[i].id && a.fileName === newQueue[i].fileName))) {
+      Logger.trace("Did not write new MatchSettings")
+      return
+    }
+
+    let maps = newQueue.map(a => `  <challenge>
+    <file>${a.fileName.replaceAll('/', '\\')}</file>
+    <ident>${a.id}</ident>
+  </challenge>
+`)
+    maps.unshift(`  <challenge>
+    <file>${curr.fileName.replaceAll('/', '\\')}</file>
+    <ident>${curr.id}</ident>
+  </challenge>
+`)
+
+    const game = tm.config.game
```

Comment:
import GameService and use GameService.config instead

Reply:
Will fix.